### PR TITLE
Interactive plots for shakemap regression/residuals

### LINKED ISSUE #1105: Add shakemap regression plot
Should be able to reuse DYFI combo plot

### LINKED ISSUE #1123: Residual plots for shakemap
Using stationlist.json

Distance options on horizontal axis:
- rrup
- rjb
- possibly other properties

Vertical axis options
- Observed values with selectable metric (PGA, PGV, MMI ...)
- Residuals

Horizontal axis should probably be log? And the vertical axis values should be logged.

Residuals should be plotted as `ln(observations) - ln(predictions)` with the plot centered at 0.

Plot DYFI converted mmi values as circles and instrumental measurements as triangles

### PR DESCRIPTION
**Pull request by dslosky-usgs**
closes #1123
closes #1105

Add getContent method to contents-xml.service to asynchronously download contents from product.contents (should eventually be used for shakemap stationlist and metadata as well as DYFI products)

Add sharedGetContent pipe to work with new getContent method on contents-xml.service

AnalysisComponent for shakemap has interactive regression/residual plots

PlotAttenPipe prepares a new shakemap product (download/attenuation_curves.json) for plotting

PlotStationsPipe prepares a shakemap station list for plotting

Make station channels description and pre-expansion optional for the shared-station component
Change station CSS class to allow the same class to be used on the map as on the charts

ng-charts changes to allow:
- multiple log axis
- symmetrical plots
- lines of symmetry
- Custom shapes for scatter plot
- custom legend icons
- better ticks

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -38,6 +38,9 @@
               {
                 "input": "node_modules/document-register-element/build/document-register-element.js"
               },
+              {
+                "input": "node_modules/document-register-element/build/document-register-element.js"
```
Comment (ehunter-usgs) [inline angular.json]: looks like this script is repeated

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "@angular/platform-browser-dynamic": "^7.2.0",
     "@angular/router": "^7.2.0",
     "@material/tab-bar": "^0.43.1",
-    "@swimlane/ngx-charts": "^10.0.0",
+    "@swimlane/ngx-charts": "github:swimlane/ngx-charts",
```
Comment (ehunter-usgs) [inline package.json]: Can we lock this into a specific version?

File: src/app/shared/get-content.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+})
+export class GetContentPipe implements PipeTransform {
+  constructor(public contentService: ContentsXmlService) {}
+
+  transform(product, content): any {
+    if (!product || !content) {
+      return null;
+    }
+
+    return this.contentService.getContent(product, content);
```
Comment (ehunter-usgs) [inline src/app/shared/get-content.pipe.ts]: Add logic from `contentService.getContent` to the pipe. Remove it from the service.

File: src/app/core/contents-xml.service.ts [hunk 4]
```diff
@@ -35,6 +37,32 @@ export class ContentsXmlService {
...
 
+  /**
+   * Download product content
+   *
+   * @param product
+   *     Product with contents property
+   * @param content
+   *     Name of the content to download
+   */
+  getContent(product: any, content: string): Observable<any> {
```
Comment (ehunter-usgs) [inline src/app/core/contents-xml.service.ts]: Remove this logic from the service. Add the http request to the pipe.

**Comment by dslosky-usgs** [comment]
@ehunter-usgs @jdbrown-USGS @cswilson-usgs 
Hey guys! Is there some more stuff I should do in here before it's ready to merge?

**Pull request merged**